### PR TITLE
AzureStack Schema and Tests

### PR DESCRIPTION
**Pull request by deathly809**
Created the schemas and tests for Microsoft.AzureStack

### CODE REVIEW

File: schemas/2015-01-01/deploymentTemplate.json [hunk 1]
```diff
@@ -480,6 +480,9 @@
                   { "$ref": "https://schema.management.azure.com/schemas/2017-09-01/Microsoft.Network.json#/resourceDefinitions/dnsZones_SOA" },
                   { "$ref": "https://schema.management.azure.com/schemas/2017-09-01/Microsoft.Network.json#/resourceDefinitions/dnsZones_SRV" },
                   { "$ref": "https://schema.management.azure.com/schemas/2017-09-01/Microsoft.Network.json#/resourceDefinitions/dnsZones_TXT" },
+                  { "$ref": "https://schema.management.azure.com/schemas/2016-03-01/Microsoft.Insights.json#/resourceDefinitions/alertrules" },
```
Comment (TianoMS) [inline schemas/2015-01-01/deploymentTemplate.json]: >{ "$ref": "https://schema.management.azure.com/schemas/2016-03-01/Microsoft.Insights.json#/resourceDefinitions/alertrules" }, [](start = 18, length = 125)

do you need this change?

File: tools/package.json [hunk 2]
```diff
@@ -10,11 +10,11 @@
...
     "chalk": "^1.1.3",
-    "sync-request": "^2.1.0",
-    "tv4": "1.2.7",
-    "vscode-json-languageservice": "3.0.2",
     "mocha": "^2.3.3",
-    "mocha-each": "^1.1.0"
+    "mocha-each": "^1.1.0",
+    "sync-request": "^2.2.0",
+    "tv4": "^1.2.7",
+    "vscode-json-languageservice": "3.0.2"
```
Comment (TianoMS) [inline tools/package.json]: why these changes?

File: tools/package-lock.json [hunk 3]
```diff
@@ -0,0 +1,204 @@
+{
+  "name": "template-schema-validation-tools",
```
Comment (TianoMS) [inline tools/package-lock.json]: do you need this file?

File: schemas/2014-04-01-preview/deploymentTemplate.json [hunk 4]
```diff
@@ -483,6 +483,8 @@
                   { "$ref": "https://schema.management.azure.com/schemas/2017-09-01/Microsoft.Network.json#/resourceDefinitions/dnsZones_SRV" },
                   { "$ref": "https://schema.management.azure.com/schemas/2017-09-01/Microsoft.Network.json#/resourceDefinitions/dnsZones_TXT" },
                   { "$ref": "https://schema.management.azure.com/schemas/2016-03-01/Microsoft.Insights.json#/resourceDefinitions/alertrules" },
+                  { "$ref": "https://schema.management.azure.com/schemas/2017-06-01-preview/Microsoft.AzureStack.json#/resourceDefinitions/registrations" },
```
Comment (TianoMS) [inline schemas/2014-04-01-preview/deploymentTemplate.json]: >2017-06-01-preview [](start = 73, length = 18)

2017-06-01?

**Comment by Tiano2017** [comment]
@deathly809 left a few comments. please take a look.

File: schemas/2015-01-01/deploymentTemplate.json (hunk 1 above)
Comment (deathly809) [reply schemas/2015-01-01/deploymentTemplate.json]: I did a pull from upstream and that was pulled in during a merge.

File: tools/package.json (hunk 2 above)
Comment (deathly809) [reply tools/package.json]: I am not sure why that was changed.  It might have to do with the fact your tools don't work right out the box.  I did an NPM install of the required packages for your stuff and it probably modified those files.  I will do a checkout of the ones from the master branch.

File: tools/package-lock.json (hunk 3 above)
Comment (deathly809) [reply tools/package-lock.json]: See above.

File: schemas/2014-04-01-preview/deploymentTemplate.json (hunk 4 above)
Comment (deathly809) [reply schemas/2014-04-01-preview/deploymentTemplate.json]: Yup.

**Comment by deathly809** [comment]
@Tiano2017  I have reverted the changes in the tools directory.

File: schemas/2014-04-01-preview/deploymentTemplate.json (hunk 4 above)
Comment (TianoMS) [reply schemas/2014-04-01-preview/deploymentTemplate.json]: This is not fixed. 

---
In reply to: [168658962](https://github.com/Azure/azure-resource-manager-schemas/pull/413#discussion_r168658962) [](ancestors = 168658962)

Comment (deathly809) [reply schemas/2014-04-01-preview/deploymentTemplate.json]: You did not tell me what was wrong. Just asked me 2017-16-01?  What does that even mean.

Comment (deathly809) [reply schemas/2014-04-01-preview/deploymentTemplate.json]: I figured out what you attempted to say.  Removed preview part of version.

File: schemas/2015-01-01/deploymentTemplate.json (hunk 1 above)
Comment (TianoMS) [inline schemas/2015-01-01/deploymentTemplate.json]: >{ "$ref": "https://schema.management.azure.com/schemas/2016-03-01/Microsoft.Insights.json#/resourceDefinitions/alertrules" }, [](start = 18, length = 125)

This is not there in master branch. I guess you didn't merge correctly. Please resolve this.
